src/include/function/list/operations/list_slice_operation.h

```diff
@@ -30,8 +30,12 @@ struct ListSlice {
 
     static inline void operation(ku_string_t& str, int64_t& begin, int64_t& end,
         ku_string_t& result, ValueVector& resultValueVector) {
-        int64_t startIdx = (begin == 0) ? 1 : begin;
-        int64_t endIdx = (end == 0) ? str.len : end;
+        int64_t startIdx = (begin == 0) ? 1 : ((begin < 0) ? (begin + str.len + 1) : begin);
+        int64_t endIdx = (end == 0) ? str.len : ((end < 0) ? (end + str.len + 1) : end);
+        if (endIdx < startIdx || startIdx < 0 || endIdx < 0) {
```

Comment:
Similar to list extraction, you seem to only modify one of the template functions. This one is for string slice but not general list slice. Make sure you modified both and test both cases. Also, don't forget to refactor/template common logic whenever possible.

A separate comment not relates to this PR (but let's change it in this PR), use substring operation to do string slice. It's always good to reuse code.